web-ui/src/components/team-results/TeamSummaryCard.jsx

```diff
@@ -1,125 +1,126 @@
-            })}
-            <br />
-            <strong>Team Members: </strong>
-            {nonLeads.map((member, index) => {
-              return index !== nonLeads.length - 1
-                ? `${member.name}, `
-                : member.name;
-            })}
-          </React.Fragment>
-        )}
-      </CardContent>
-      <CardActions>
-        <SplitButton options={options} onClick={handleAction} />
-      </CardActions>
-      <EditTeamModal
-        team={team}
-        open={open}
-        onClose={handleClose}
-        onSave={(team) => {
-          const copy = [...teams];
-          copy[index] = team;
-          dispatch({
-            type: UPDATE_TEAMS,
-            payload: copy,
-          });
-          handleClose();
-        }}
-      />
-    </Card>
-  );
+const TeamSummaryCard = ({team, index}) => {
+    const {state, dispatch} = useContext(AppContext);
+    const {teams, userProfile, csrf} = state;
+    const [open, setOpen] = useState(false);
+    const isAdmin =
+        userProfile && userProfile.role && userProfile.role.includes("ADMIN");
+
+    let leads =
+        team.teamMembers == null
+            ? null
+            : team.teamMembers.filter((teamMember) => teamMember.lead);
+    let nonLeads =
+        team.teamMembers == null
+            ? null
+            : team.teamMembers.filter((teamMember) => !teamMember.lead);
+
+    const isTeamLead =
+        leads === null
+            ? false
+            : leads.some((lead) => lead.memberid === userProfile.memberProfile.id);
+
+    const handleOpen = () => setOpen(true);
+
+    const handleClose = () => setOpen(false);
+
+    const deleteATeam = async (id) => {
+        if (id && csrf) {
+            const result = await deleteTeam(id, csrf);
+            if (result && result.payload && result.payload.status === 200) {
+                window.snackDispatch({
+                    type: UPDATE_TOAST,
+                    payload: {
+                        severity: "success",
+                        toast: "Team deleted",
+                    },
+                });
+                let newTeams = teams.filter((team) => {
+                    return team.id !== id;
+                });
+                dispatch({
+                    type: UPDATE_TEAMS,
+                    payload: newTeams,
+                });
+            }
+        }
+    };
+
+    const options =
+        isAdmin || isTeamLead ? ["Edit Team", "Delete Team"] : ["Edit Team"];
+
+    const handleAction = (e, index) =>
+        index === 0 ? handleOpen() : deleteATeam(team.id);
+
+    return (
+        <Card className="summary-card">
+            <CardHeader title={team.name} subheader={team.description}/>
+            <CardContent>
+
+                {team.teamMembers == null ? (
+                    <React.Fragment>
+                    <strong>Team Leads: </strong>
+                    <br/>
+                    <strong>Team Members: </strong>
```

Comment:
let's have this print "None Assigned" when there are no members or leads rather than just not printing anything.

Reply:
right....sorry, forgot. _gets right on it_

Reply:
Done